### DISCUSSION TITLE
is it possible to always use nicegui's widgets in slot directly?

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#2080] [Ideas / Feature Requests]
This is a following thought of [this discussion](https://github.com/zauberzeug/nicegui/discussions/2041)

@falkoschindler shows a general way to add slot, i.e, using quasar code
```python
tabRange.add_slot('body-cell-percent', r'''
    <q-td :props="props">
        <q-linear-progress size="1rem" :value="props.value" color="red" track-color="green">
            <div class="absolute-center text-black text-sm">
                {{ (props.value * 100).toFixed(0) }}%
            </div>
        </q-linear-progress>
    </q-td>
''')
```
which is some difficult to write for user, like me, knows little about VUE/Quasar.

Since nicegui has supplied `ui.linear_progress`(well, based on `q-linear-progress`), what comes into my mind naturally is 
```python
with tabRange.add_slot('body-cell-percent'):
    ui.linear_progress().props('value=0.1').classes('w-full')
```
but as we may find that, we can't sets every cell's value to every `ui.linear_progress`.

So any way to move more quasar code out of projects based on nicegui by using pre-defined widgets? Thanks

**Comment by falkoschindler** [comment]
Let me refer to a comment in a previous discussion https://github.com/zauberzeug/nicegui/discussions/1138#discussioncomment-6383197 which still holds today:

> These scoped slots are certainly one of the most complicated things NiceGUI has to offer. I'm thinking about ways to simplify it for some time now, but so far I don't see how. Since these elements q-tr, q-td, ... don't live in Python (they are no NiceGUI elements), we would need something new to define template objects. It would have to allow for all possible HTML tags and attributes. And we need the ability access row props and to handle and forward events. It all needs to be nice and simple enough to justify the implementation and learning this new API instead of learning to write Vue templates. Pretty complicated, but maybe not impossible. 🙂

So I'm still pretty clueless how we could improve the slot API. But I'm open for any suggestions.

**Comment by falkoschindler** [comment]
This feature has just been released via PR #5592 in NiceGUI 3.5.0.
